### PR TITLE
cool-retro-term 1.1.1 -> 1.2.0

### PR DESCRIPTION
**Pull request by brendon-boldt**
###### Description of changes

First-time PR; reviewed the contribution notes on GitHub and the manual carefully but still might have missed something.

Update `cool-retro-term` from `1.1.1` to `1.2.0` primarily for the reason that `Ctrl-Space`/`^@` was not working in `1.1.1` and was fixed in the new version.  It was also necessary to update `qmltermwidget` to fix the `cool-retro-term` build.

[Release notes for `cool-retro-term`](https://github.com/Swordfish90/cool-retro-term/releases/tag/1.2.0).  `Swordfish90/qmltermwidget` does not have any release notes (just a commit is referenced in the git submodule), but the upstream `lxqt/qmltermwidget` has [release notes](https://github.com/lxqt/qtermwidget/releases) (`0.9.0` to the most recent are relevant to this update).

Testing
- `nix-build ../nixpkgs/ -A cool-retro-term.passthru.tests.test`
- Ran the newly compiled binary briefly without issue

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by brendon-boldt** [comment]
I think this can be backported, but I do not see a way to add the label/tag.

File: pkgs/applications/terminal-emulators/cool-retro-term/default.nix [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
     owner = "Swordfish90";
     repo = "cool-retro-term";
     rev = version;
-    sha256 = "0mird4k88ml6y61hky2jynrjmnxl849fvhsr5jfdlnv0i7r5vwi5";
+    sha256 = "PewHLVmo+RTBHIQ/y2FBkgXsIvujYd7u56JdFC10B4c=";
   };
 
   patchPhase = ''
     sed -i -e '/qmltermwidget/d' cool-retro-term.pro
   '';
```
Comment (tpwrules) [inline pkgs/applications/terminal-emulators/cool-retro-term/default.nix:17]: I would replace this with the following as a patch file. That way the intent is clear and other patch files can be added. Replacing phases completely like this causes problems in overrides.

`use-system-qmltermwidget.patch`

```
diff --git a/cool-retro-term.pro b/cool-retro-term.pro
index 851f0b4..b24c473 100644
--- a/cool-retro-term.pro
+++ b/cool-retro-term.pro
@@ -2,7 +2,6 @@ TEMPLATE = subdirs
 
 CONFIG += ordered
 
-SUBDIRS += qmltermwidget
 SUBDIRS += app
 
 desktop.files += cool-retro-term.desktop
```

File: pkgs/applications/terminal-emulators/cool-retro-term/default.nix [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
 
-  buildInputs = [ qtbase qtquick1 qmltermwidget qtquickcontrols qtgraphicaleffects ];
+  buildInputs = [ qtbase qtquick1 qmltermwidget qtquickcontrols2 qtgraphicaleffects ];
```
Comment (tpwrules) [inline pkgs/applications/terminal-emulators/cool-retro-term/default.nix:19]: This still needs `qtquickcontrols` on Darwin (in addition to `qtquickcontrols2`) or I get the error `qrc:/SettingsWindow.qml:25:1: module "QtQuick.Dialogs" is not installed`.

Not sure if this is a defect in Nixpkgs or Qt.

**Review by tpwrules — CHANGES_REQUESTED** [review]
This looks good on your part. I tested on aarch64-darwin and ran into an issue.

There are also issues with ugly rendering on the official Nvidia drivers and Linux. There doesn't appear to be a fix available from upstream yet. Maybe this should wait until then.

Especially considering the Nvidia driver and that there is no brokenness in the current version, I would veto backporting this.

File: pkgs/development/libraries/qmltermwidget/default.nix [hunk 3]
```diff
@@ -1,28 +1,20 @@
 { lib, stdenv, fetchFromGitHub, qtbase, qtquick1, qmake, qtmultimedia, utmp, fetchpatch }:
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qmltermwidget/default.nix:1]:
```suggestion
{ lib, stdenv, fetchFromGitHub, qtbase, qtquick1, qmake, qtmultimedia, utmp }:
```

File: pkgs/development/libraries/qmltermwidget/default.nix [hunk 4]
```diff
@@ -1,28 +1,20 @@
...
 
 stdenv.mkDerivation {
-  version = "2018-11-24";
+  version = "2022-09-16";
   pname = "qmltermwidget-unstable";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qmltermwidget/default.nix:5]:
```suggestion
  pname = "qmltermwidget-unstable";
  version = "unstable-2022-09-16";
```

File: pkgs/applications/terminal-emulators/cool-retro-term/default.nix [hunk 5]
```diff
@@ -1,22 +1,22 @@
...
 
   src = fetchFromGitHub {
     owner = "Swordfish90";
     repo = "cool-retro-term";
     rev = version;
-    sha256 = "0mird4k88ml6y61hky2jynrjmnxl849fvhsr5jfdlnv0i7r5vwi5";
+    sha256 = "PewHLVmo+RTBHIQ/y2FBkgXsIvujYd7u56JdFC10B4c=";
   };
 
   patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/cool-retro-term/default.nix:15]:
```suggestion
  postPatch = ''
```

File: pkgs/applications/terminal-emulators/cool-retro-term/default.nix [hunk 6]
```diff
@@ -1,22 +1,22 @@
 { lib, stdenv, fetchFromGitHub, mkDerivation, qtbase, qtquick1, qmltermwidget
-, qtquickcontrols, qtgraphicaleffects, qmake, nixosTests }:
+, qtquickcontrols2, qtgraphicaleffects, qmake, nixosTests }:
 
 mkDerivation rec {
-  version = "1.1.1";
+  version = "1.2.0";
   pname = "cool-retro-term";
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/cool-retro-term/default.nix:6]:
```suggestion
  pname = "cool-retro-term";
  version = "1.2.0";
```

File: pkgs/applications/terminal-emulators/cool-retro-term/default.nix (hunk 2 above)
Comment (brendon-boldt) [reply pkgs/applications/terminal-emulators/cool-retro-term/default.nix]: Added `qtquickcontrols` in as a conditional dependency for Darwin.

File: pkgs/applications/terminal-emulators/cool-retro-term/default.nix (hunk 1 above)
Comment (brendon-boldt) [reply pkgs/applications/terminal-emulators/cool-retro-term/default.nix]: Refactored as suggested; thank you.

File: pkgs/applications/terminal-emulators/cool-retro-term/default.nix [hunk 7]
```diff
@@ -1,22 +1,21 @@
...
   };
 
-  patchPhase = ''
-    sed -i -e '/qmltermwidget/d' cool-retro-term.pro
-  '';
+  patches = [ ./use-system-qmltermwidget.patch ];
 
-  buildInputs = [ qtbase qtquick1 qmltermwidget qtquickcontrols qtgraphicaleffects ];
+  buildInputs = [ qtbase qtquick1 qmltermwidget qtquickcontrols2
+    qtgraphicaleffects ] ++ lib.optional stdenv.isDarwin qtquickcontrols;
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/cool-retro-term/default.nix:18]:
```suggestion
  buildInputs = [
    qtbase
    qtquick1
    qmltermwidget
    qtquickcontrols2
    qtgraphicaleffects
  ] ++ lib.optional stdenv.isDarwin qtquickcontrols;
```
this was a bit weird, lets do it properly.

**Review by SuperSandro2000 — APPROVED** [review]
other than that LGTM

File: pkgs/applications/terminal-emulators/cool-retro-term/default.nix (hunk 7 above)
Comment (brendon-boldt) [reply pkgs/applications/terminal-emulators/cool-retro-term/default.nix:18]: I'll make this change -- is there a style guide or formatter for nix lang, by the way?

**Comment by brendon-boldt** [comment]
@tpwrules Would it be worth it to try an earlier commit of `qmltermwidget` to see if it fixes the Nvidia issue?  If you think so, I can get you that to test out.

**Comment by tpwrules** [comment]
Some context: https://github.com/Swordfish90/cool-retro-term/issues/703#issuecomment-1058518374

I highly doubt changing `qmltermwidget` will fix it.

File: pkgs/applications/terminal-emulators/cool-retro-term/default.nix (hunk 7 above)
Comment (SuperSandro2000) [reply pkgs/applications/terminal-emulators/cool-retro-term/default.nix]: https://github.com/nix-community/nixpkgs-fmt

**Comment by MinerSebas** [comment]
This https://github.com/Swordfish90/cool-retro-term/issues/739#issuecomment-1347784249 describes a change that fixes the ugly rendering with the nvidia driver.

**Comment by SuperSandro2000** [comment]
this PR got replaced by #244499

**Pull request closed without merging**